### PR TITLE
feat(console): Revamp of IdP Certificates upload and JWKS input form

### PR DESCRIPTION
**Pull request by yasinmiran**
### Related Issues
- Issue `#1` or (None)

### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- Related PR `#1` or (None)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/forms/general-details-form.tsx [hunk 1]
```diff
@@ -250,31 +262,24 @@ export const GeneralDetailsForm: FunctionComponent<GeneralDetailsFormPopsInterfa
...
-                && <>
-                    < Divider hidden />
+            { identityProviderConfig.generalDetailsForm.showCertificate && (
+                <React.Fragment>
+                    < Divider hidden/>
                     <Grid>
                         <Grid.Row columns={ 1 }>
                             <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 16 }>
-                                <Heading as="h4">Certificates</Heading>
+                                <Heading as="h4" style={ { fontWeight: 600 } }>Certificates</Heading>
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/forms/general-details-form.tsx]: Addressed with commit 36f6e44

File: apps/console/src/features/identity-providers/components/settings/idp-certificates/add-idp-certificate-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+    const [ alert, setAlert, alertComponent ] = useWizardAlert();
+
+    const [ pemBase64String, setPemBase64String ] = useState<string>("");
+    const [ submitShouldBeDisabled, setSubmitShouldBeDisabled ] = useState<boolean>(true);
+    const [ requestInProgress, setRequestInProgress ] = useState<boolean>(false);
+
+    // Event handlers
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/settings/idp-certificates/add-idp-certificate-modal.tsx]: Addressed with commit 3688ef8

File: apps/console/src/features/identity-providers/components/settings/idp-certificates/empty-certificates-placeholder.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                data-testid={ `${ testId }-empty-placeholder` }
+            />
+        </Segment>
+    );
+
+};
+
+// Component constants.
+
+const AUTH_PROV_PLACEHOLDER_EMPTY_I18N_KEY = "console:develop.features.authenticationProvider" +
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/settings/idp-certificates/empty-certificates-placeholder.tsx]: Addressed with commit 978298e

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx]: Addressed in commit c360648

File: apps/console/src/features/identity-providers/components/forms/general-details-form.tsx [hunk 5]
```diff
@@ -81,6 +81,16 @@ interface GeneralDetailsFormPopsInterface extends TestableComponentInterface {
      * Specifies if the component should only be read-only.
      */
     isReadOnly: boolean;
+    /**
+     * Explicitly specifies whether the currently displaying
+     * IdP is a SAML provider or not.
+     */
+    isSaml?: boolean;
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/forms/general-details-form.tsx:88]: Will move all these boolean check props to enums in a latter PR.

**Review by brionmario — APPROVED** [review]

**Pull request merged**